implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpDistributionSummary.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.registry.otlp;
+
+import io.micrometer.common.lang.Nullable;
+import io.micrometer.core.instrument.AbstractDistributionSummary;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.StepHistogram;
+import io.micrometer.core.instrument.step.StepMax;
+import io.micrometer.core.instrument.step.StepTuple2;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.DoubleAdder;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * A {@link io.micrometer.core.instrument.DistributionSummary} implementation customised
+ * for OTLP protocol to export data based on the
+ * {@link io.opentelemetry.proto.metrics.v1.AggregationTemporality} configured. In case of
+ * Delta aggregation, this uses a {@link StepMax} and {@link StepHistogram} to measure the
+ * distribution.
+ *
+ * @author Tommy Ludwig
+ * @author Lenin Jaganathan
+ */
+class OtlpDistributionSummary extends AbstractDistributionSummary implements StartTimeAwareMeter {
```

Comment:
Was the motivation of combining to one OtlpDistributionSummary class (getting rid of the cumulative and step specific classes) just to reduce code duplication? I think we may be able to achieve that in other ways, and it is probably cleaner in the end to have two separate classes that have different concerns, rather than try to implement both behaviors in one class, even though only one behavior will be used at a time.

Reply:
That is one thing. But if you see I had implemented the AbstractTimer directly and added some changes to the abstract timer to accept a histogram implementation directly. This will be helpful not only in Otlp but in implementations like Prometheus and SignalFx where implementors can provide a custom Histogram implementation while extending AbstractTimer instead of creating a duplicate object.

Reply:
@shakuzen What are your thoughts on this? I was initially advocating for Do-One thing behaviour and that's the reason I introduced 2 different implementations in last PR. But in this PR I have added certain this which makes sense to have a common implementation. The major reason to provide a custom max implementation and also to use the new Abstract Timer constructor to pass an Histogram implementation which can be helpful not just for OTLP but for other registries too. 

I am not sure if Cumulative/Step Timers can take custom histogram implementations. But either way, DeltaTimer needs to have a custom max implementation, so it needs implement Abstract at this point.

Reply:
I still feel like it would probably be cleaner to keep the concerns separate rather than mix different concerns in one class.

> I am not sure if Cumulative/Step Timers can take custom histogram implementations.

It would require adding constructors to those, but they can pass a histogram implementation to the super AbstractTimer constructor.

> DeltaTimer needs to have a custom max implementation, so it needs implement Abstract at this point.

I see the difficulty in overriding the max implementation now. We could introduce a new interface that both max implementations can implement and then it can be swappable. What do you think?

Reply:
>  We could introduce a new interface that both max implementations can implement and then it can be swappable. What do you think?

Yeah, I wanted to take that route to introduce a High-level interface(even Higher than max) something that defines 2 methods record() and poll() which can be used for any statistic say we need to incorporate in OTLP or SD for some other implementations.

Wondering does it make the Step/Cumulative objects too customizable. Also, since StepTimer accepts a WindowMax we cannot enforce that it can accept only StepValue.

Reply:
I agree with Tommy here, could you please revert this back and keep them separate?
I thought on the previous PR we decided to keep them separated (and deal with the dupe). Since they are internal components we will try to optimize them later if we will feel the need.

Reply:
I brought back the Cumulative and Step (maybe it should be Delta) flavors with the latest commit.